datadog_checks_dev/tests/test_docker.py

```diff
@@ -69,8 +76,9 @@ def test_retry_on_failed_conditions(self, attempts, expected_call_count):
                 expected_exception = Exception
 
         with pytest.raises(expected_exception):
-            with docker_run(compose_file, attempts=attempts, conditions=[condition]):
-                pass
+            with patch('datadog_checks.dev.docker.ComposeFileUp'), patch('datadog_checks.dev.docker.ComposeFileDown'):
+                with docker_run(compose_file, attempts=attempts, conditions=[condition], attempts_wait=0):
+                    pass
```

Comment:
These nested `with`s might look a bit cleaner using the `with A, B:` syntax. What do you think?

An alternative that might make sense here is to avoid the patching here and passing the `up` and `down` mock functions explicitly to `docker_run` (paired with `compose_file=None`).

Reply:
I did both, that's a good idea. It made the test surprisingly easier in fact